docs/contributing/Style-guide-for-Flutter-repo.md

```diff
@@ -1028,32 +1028,17 @@ to actually desugar the generator into something that uses an iterator class.
 Embrace code duplication in tests. It makes it easier to make new tests by copying and pasting them and
 tweaking a few things.
 
-Avoid using `setUp`, `tearDown`, and similar features, as well as test-global variables or other state
-shared between tests. They make writing tests easier but make maintaining them, debugging them, and
-refactoring code much harder. (These are commonly used in Flutter's codebase today, but that is almost
-always a mistake. When you are editing a file that uses those features, aim to reduce the number of
-tests using them while you're there.)
+Avoid using test-global variables or other state shared between tests. They make writing tests easier
```

Comment:
The `setUp` and `tearDown` are used in many places currently.
Should we fix that? Or remove this specification from the style guide?

Reply:
I'm a big believer in stateless or near-stateless test fixtures, and have been chipping away at the pattern in flutter/packages. I often point people to https://abseil.io/tips/122 in reviews (I forgot we had this here).

I still think this is good advice, personally (and will continue to advocate for it even if it's not in the guide).